### PR TITLE
Allow containers to be named and assigned to a network on startup

### PR DESCRIPTION
**Pull request by thomaseizinger**
Docker::run_args allows users to specify a docker network and a name
for the container. The network will be created on-demand and clean-up
in the end when the client is dropped.

Co-authored-by: Aleksandrov Vladimir <invis87@gmail.com>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,13 +7,17 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Added
+
+-   `Docker#run_with_args` method. This allows naming a container and assigning it to a specific docker network. The docker
```
Comment (D4nte) [inline CHANGELOG.md:12]:
```suggestion
-   `Docker::run_with_args` method. This allows naming a container and assigning it to a specific docker network. The docker
```
Why using a hash symbol?

File: src/clients/cli.rs [hunk 2]
```diff
@@ -195,6 +223,53 @@ impl Docker for Cli {
     }
 }
 
+impl Drop for Cli {
```
Comment (D4nte) [inline src/clients/cli.rs:226]: Are you testing that dropping the cli stops the container?

**Review by D4nte — APPROVED** [review]

**Comment by goller** [comment]
Just tried the branch; works great !

File: src/clients/cli.rs (hunk 2 above)
Comment (thomaseizinger) [reply src/clients/cli.rs:226]: Dropping the client does not stop the container, dropping the container stops the container :)

Rust lifetimes guarantee that the containers are dropped before the client.

File: CHANGELOG.md (hunk 1 above)
Comment (thomaseizinger) [reply CHANGELOG.md:12]: Because I've wrote too much Java in my life.

**Pull request merged**